dynamiqs/mesolve/mesolve.py

```diff
@@ -182,11 +182,11 @@ def mesolve(
     jump_ops = to_tensor(jump_ops, dtype=options.cdtype, device=options.device)
 
     # convert tsave to a tensor
-    tsave = to_tensor(tsave, dtype=options.rdtype, device=options.device)
+    tsave = to_tensor(tsave, dtype=options.rdtype, device='cpu')
```

Comment:
Since we use `tsave` as a numpy array (in order to keep floats everywhere), shouldn't we convert it to a numpy array from the get-go, instead of calling `.numpy()` slightly later in the code? 

Or otherwise we can move `tsave` to the GPU and never call `.cpu.numpy()`. This was my preferred solution and still is, from this debate we had some time ago. Just mentionning it in case you changed your mind.

Reply:
Yes we should, I was just lazy to recode the following line `check_time_tensor` for NumPy arrays.

> Or otherwise we can move tsave to the GPU and never call .cpu.numpy(). This was my preferred solution and still is, from this debate we had some time ago. Just mentionning it in case you changed your mind.
Let me think about it, I propose to keep this aside for now and choose either fix later (make times Tensor everywhere, or convert to numpy directly).

Reply:
Sound good !